napari/layers/points/points.py

```diff
@@ -806,19 +796,38 @@ def current_symbol(self, symbol: Union[None, float]) -> None:
 
     @property
     def size(self) -> np.ndarray:
-        """(N, D) array: size of all N points in D dimensions."""
+        """(N,) array: size of all N points."""
         return self._size
 
     @size.setter
     def size(self, size: Union[int, float, np.ndarray, list]) -> None:
         try:
-            self._size = np.broadcast_to(size, self.data.shape).copy()
+            self._size = np.broadcast_to(size, len(self.data)).copy()
         except ValueError as e:
-            try:
-                self._size = np.broadcast_to(
-                    size, self.data.shape[::-1]
-                ).T.copy()
-            except ValueError:
+            if (
```

Comment:
I don't love the increase in complexity here, especially given that this PR is simplifying `Points.size`. Eventually we have a `try` block in an `if` block in a `try` block.

Do we really need to handle this special failure of broadcasting? It seems like we just need to handle scalar value, a vector value, then maybe gracefully handle the old matrix value, then fail if the value is anything else.

Reply:
The reason I did it this way is because, in order to handle the old matrix value, we have to first ensure it's an array to check its shape, so `np.broadcast` serves that purpose too, in case it's a nested list, for example.

How would you do it otherwise? Would you warn or not in case of anisotropic value passed?

Reply:
Actually, this is also accounted for :)